src/core/ext/filters/client_channel/resolver/xds/xds_resolver.cc

```diff
@@ -803,20 +826,16 @@ XdsResolver::XdsConfigSelector::GetCallConfig(GetCallConfigArgs args) {
     call_config.service_config = std::move(method_config);
   }
   call_config.call_attributes[XdsClusterAttribute::TypeName()] =
-      args.arena->New<XdsClusterAttribute>(it->first);
+      args.arena->New<XdsClusterAttribute>(cluster->cluster_name());
   std::string hash_string = absl::StrCat(hash.value());
   char* hash_value =
       static_cast<char*>(args.arena->Alloc(hash_string.size() + 1));
   memcpy(hash_value, hash_string.c_str(), hash_string.size());
   hash_value[hash_string.size()] = '\0';
   call_config.call_attributes[RequestHashAttribute::TypeName()] =
       args.arena->New<RequestHashAttribute>(hash_value);
-  call_config.on_commit = [cluster_state = it->second->Ref()]() mutable {
-    cluster_state.reset();
-  };
-  auto lb_data =
-      args.arena->ManagedNew<XdsClusterLbDataAttribute>(cluster_map_);
-  call_config.call_attributes[XdsClusterLbDataAttribute::TypeName()] = lb_data;
+  auto lb_data = args.arena->ManagedNew<XdsClusterDataAttribute>(cluster_map_);
```

Comment:
Please call this variable `cluster_map`.  It actually has nothing to do with the LB policies.